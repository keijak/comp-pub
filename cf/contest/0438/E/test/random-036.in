280 117
41
597
530
243
500
66
241
488
938
820
294
382
411
217
362
943
455
909
851
962
265
395
18
61
305
977
729
85
348
568
380
678
887
262
890
844
792
346
496
975
808
926
446
410
535
520
659
954
679
875
635
359
529
988
186
502
405
199
884
36
452
42
367
431
755
846
740
900
644
80
600
1000
662
950
514
386
618
946
281
749
170
46
605
137
184
769
132
882
632
216
723
609
614
984
207
44
99
986
192
714
342
72
869
145
800
985
839
734
598
579
491
945
387
992
87
483
607
208
990
761
179
283
313
656
782
719
263
536
677
254
812
964
270
57
22
866
493
282
147
497
160
108
375
623
355
302
752
833
228
196
100
276
867
31
698
253
762
320
910
416
779
307
19
347
994
23
617
43
436
570
933
331
119
878
138
322
83
17
573
717
238
899
203
972
534
291
722
578
490
592
516
384
158
660
421
807
78
822
318
687
849
784
710
648
445
279
242
957
267
77
705
757
366
604
630
58
613
374
341
329
681
176
628
388
397
776
731
642
463
588
30
188
319
684
916
532
725
377
650
689
503
696
862
612
5
278
871
594
547
941
823
853
756
398
682
921
419
52
473
980
561
101
468
8
122
229
440
783
327
62
585
798
970
971
130
112
505
180
693
501
