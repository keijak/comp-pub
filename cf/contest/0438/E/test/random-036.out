0
0
0
0
1
0
0
1
0
2
0
0
4
0
5
2
1
16
1
14
15
5
61
9
46
88
23
229
71
167
455
115
878
493
650
2189
648
3458
3056
2701
10146
3928
14002
17431
12032
46156
24085
58365
93854
57787
208389
144444
251058
485654
297243
940616
838196
1119232
2445657
1606108
4267464
4707497
5194439
12094274
8909203
19551099
25695668
25145323
59130473
49804879
90857032
137018288
126619014
287351287
277352598
430066045
717268352
658508169
396038034
531276264
83705886
707703840
509011631
792731483
348782745
315419374
950657214
927386338
250169695
953997131
238253222
206624858
688166396
227915020
429558667
901244133
706337954
388020793
521636669
442740036
278702919
962989056
16977420
558937760
228502324
243624278
202610068
278339977
85045776
710937995
259893057
6846581
818531263
77015075
522100897
947318803
652333924
